cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaCluster.java

```diff
@@ -658,6 +662,18 @@ public static KafkaCluster fromCrd(Kafka kafkaAssembly, KafkaVersion.Lookup vers
         return result;
     }
 
+    private static void validateConfigProperty(String propertyName, KafkaClusterSpec kafkaClusterSpec) {
+        String orLess = kafkaClusterSpec.getReplicas() > 1 ? " or less" : "";
+        if (kafkaClusterSpec.getConfig() != null) {
+            if (kafkaClusterSpec.getConfig().get(propertyName) != null) {
+                if (Integer.parseInt(kafkaClusterSpec.getConfig().get(propertyName).toString()) > kafkaClusterSpec.getReplicas()) {
```

Comment:
Unless there's already been some validation of the config by this point, then you can't assume that the value of the `propertyName` is parseable as an `int`, so you need to catch `NumberFormatException`.